src/pokemon/dexService.ts

```diff
@@ -0,0 +1,40 @@
+import fs = require("fs");
+import FuzzyMatching = require('fuzzy-matching');
+import { toID } from "../app";
+import { Pokemon, Dex } from "./models";
+
+export class DexService {
+
+	private pokemonMap: { [name: string]: Pokemon } = {};
+	private dex: Dex = {} as Dex;
+	private fuzzyMatching: FuzzyMatching;
+
+	constructor() {
+		this.loadFileData();
+		this.fuzzyMatching = new FuzzyMatching(this.dex.pokemon.map(p => p.name));
+	}
+
+	public getPokemon(name: string): Pokemon | undefined {
```

Comment:
```suggestion
	public getPokemon(name: string): Pokemon? {
```
Pretty sure this works in type returns.

Reply:
It didn't compile for me. I'm using the TypeScript 3.6.3, the one we have setup for the solution. Perhaps I'm missing something here. =/

Reply:
Alright, we can leave it then